packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx

```diff
@@ -130,7 +130,7 @@ export const CheckboxGroupInput: FunctionComponent<CheckboxGroupInputProps> = pr
         );
     }
 
-    if (!isLoading && allChoices === undefined) {
+    if (!isLoading && !fetchError && allChoices === undefined) {
```

Comment:
Maybe worth doing this in the `RadioButtonGroupInput` and `SelectInput` components as well

Reply:
I've tested that it's not necessary for these